### PR TITLE
GPC-104: Add cddo openapi linting

### PR DESCRIPTION
**Pull request by oskarwilliams378**

### CODE REVIEW

File: .github/workflows/openapi-lint.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          node-version: 18
+
+      - name: Spin up localstack and db
+        uses: yu-ichiro/spin-up-docker-compose-action@v1
+        with:
+          file: docker-compose-local.yml
+
+      - name: Install packages
+        run: npm ci
```
Comment (olevett) [inline .github/workflows/openapi-lint.yml:33]: Does the generate step need the npm dependency, if not, I'd be tempted to put it down a bit closer to the test step

File: package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "devDependencies": {
+    "@stoplight/spectral-cli": "^6.2.0",
```
Comment (olevett) [inline package.json:3]: I think, given we've got dependabot etc, I'd make this a fixed dependency and lose the `^`, not too strongly opinionated here though

File: .github/workflows/openapi-lint.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Generate API docs
+        run: ./gradlew -Pdocs generateOpenApiDocs
+
+      - name: Lint API docs
+        run: npm run openapi-lint
```
Comment (olevett) [inline .github/workflows/openapi-lint.yml:39]: Do we get any concrete output of this, or is it just a pass/fail?

File: package.json (hunk 2 above)
Comment (oskarwilliams378) [reply package.json]: Sure

File: .github/workflows/openapi-lint.yml (hunk 1 above)
Comment (oskarwilliams378) [reply .github/workflows/openapi-lint.yml]: Will do

File: package.json (hunk 2 above)
Comment (ethanmills) [reply package.json]: do we need to configure dependabot for package.json?

Comment (oskarwilliams378) [reply package.json]: looks like we may not need package at all I hope

File: .github/workflows/openapi-lint.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Spin up localstack and db
+        uses: yu-ichiro/spin-up-docker-compose-action@v1
+        with:
+          file: docker-compose-local.yml
+
+      - name: Generate API docs
+        run: ./gradlew -Pdocs generateOpenApiDocs
+
+      - name: Lint API docs
+        uses: stoplightio/spectral-action@latest
```
[the remark this answers is not in the document]
Comment (oskarwilliams378) [reply .github/workflows/openapi-lint.yml:31]: I'm getting it working then will lock it down

**Review by olevett — APPROVED** [review]

**Pull request merged**